src/clients.rs

```diff
@@ -274,12 +310,16 @@ impl StateClient for StateStore {
 
             let note = Note::from_slice(&rsp.note)?;
 
-            for (ssk, vk, psk) in addresses.iter() {
+            for (index, (vk, psk)) in self.addresses.borrow().iter().enumerate()
+            {
                 if vk.owns(&note) {
+                    let index = index as u64;
+                    let ssk = self.store.retrieve_ssk(index)?;
```

Comment:
No need to access to the store at every note. I'm for remove the `self.addresses` at all and collect them at the begin of this method (as it was before this PR)

Reply:
Not accessing it at every note but every owned note. If we resort to the before approach, how can we populate the Column families without running it everytime we fetch_notes? We don't wanna insert column families every time we fetch_notes